ttnn/cpp/ttnn/operations/data_movement/reshape_view/reshape.hpp

```diff
@@ -5,19 +5,36 @@
+
 
 namespace ttnn {
 namespace operations::data_movement {
+namespace detail {
+    ttnn::Tensor host_reshape(const ttnn::Tensor& tensor, const ttnn::Shape& shape);
+    ttnn::Tensor convert_tensor_to_rm_reshape_convert_back_to_orig_layout(const ttnn::Tensor& tensor, const ttnn::Shape& shape, const uint32_t tile_first_dim, const uint32_t tile_second_dim, const MemoryConfig &memory_config, const uint8_t queue_id, const PadValue &pad_value);
+    ttnn::Tensor fix_shape_and_perform_reshape_on_2D_RM(const ttnn::Tensor& tensor, const ttnn::Shape& shape, const uint32_t tile_first_dim, const uint32_t tile_second_dim, const MemoryConfig &memory_config, const uint8_t queue_id);
+    ttnn::Tensor fix_shape_and_perform_reshape_on_3D_TILE( const ttnn::Tensor& tensor, const ttnn::Shape& shape, const uint32_t tile_first_dim, const uint32_t tile_second_dim, const MemoryConfig &memory_config, const uint8_t queue_id, const PadValue &pad_value);
+    ttnn::Tensor perform_reshape_on_2D_RM(const ttnn::Tensor& tensor, const ttnn::Shape& shape, const MemoryConfig &memory_config, const uint8_t queue_id);
+    ttnn::Tensor convert_tile_to_rm(const ttnn::Tensor& tensor, const ttnn::Shape& shape, const uint32_t tile_first_dim, const uint32_t tile_second_dim, const MemoryConfig &memory_config, const uint8_t queue_id, const PadValue &pad_value);
+}
+
+ttnn::Shape tiling_reshape_corrector(const ttnn::Shape& shape);
+ttnn::Tensor PerformView(const ttnn::Tensor& tensor, const ttnn::Shape& shapeconst, uint32_t tile_first_dim, const uint32_t tile_second_dim);
+void Validate_transform (const ttnn::Shape& input_shape, const ttnn::Shape& output_shape);
 
 struct ReshapeViewOperation {
+    static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, const ttnn::Shape& shape,const std::optional<MemoryConfig> &memory_config,const uint8_t queue_id,const std::optional<PadValue> &pad_value);
+    static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, const ttnn::SimpleShape& logical_shape,const std::optional<MemoryConfig> &memory_config,const uint8_t queue_id,const std::optional<PadValue> &pad_value);
+    static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, tt::stl::Span<const int32_t> shape_vector,const std::optional<MemoryConfig> &memory_config,const uint8_t queue_id,const std::optional<PadValue> &pad_value);
     static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, const ttnn::Shape& shape);
     static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, const ttnn::SimpleShape& logical_shape);
     static ttnn::Tensor invoke(const ttnn::Tensor& input_tensor, tt::stl::Span<const int32_t> shape_vector);
 };
 
+
 }  // namespace operations::data_movement
 
-constexpr auto reshape =
-    ttnn::register_operation<"ttnn::reshape", ttnn::operations::data_movement::ReshapeViewOperation>();
+constexpr auto reshape = ttnn::register_operation<"ttnn::reshape", ttnn::operations::data_movement::ReshapeViewOperation>();
```

Comment:
want to doublecheck. I did not see `launch_op`, should this be `register_operation_with_auto_launch_op`?

Reply:
I have operation::run in reshape.cpp line 246. I don't always run the device part of the operation since metadata only changes do not go on device, I want to be able to control if the operation will be run or not. Maybe my understanding of how operation registering is wrong?